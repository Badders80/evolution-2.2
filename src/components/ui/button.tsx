import * as React from "react";
import { Slot } from "@radix-ui/react-slot";
import { cva, type VariantProps } from "class-variance-authority";
import { cn } from "@/lib/utils";

const buttonVariants = cva(
  "inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none",
  {
    variants: {
      variant: {
        default: "bg-gold text-black hover:bg-gold/80",
        outline: "border border-gold text-gold hover:bg-gold hover:text-black",
        ghost: "text-gold hover:bg-gold/10",
        glow: "bg-gold text-black shadow-lg shadow-gold/50 hover:shadow-xl hover:shadow-gold/70",
      },
      size: {
        default: "px-4 py-2",
        sm: "px-3 py-1 text-sm",
        lg: "px-6 py-3 text-lg",
      },
    },
    defaultVariants: {
      variant: "default",
      size: "default",
    },
  }
);

export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement>,
    VariantProps<typeof buttonVariants> {
  asChild?: boolean;
}

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  ({ className, variant, size, asChild = false, ...props }, ref) => {
    const Comp = asChild ? Slot : "button";
    return (
      <Comp className={cn(buttonVariants({ variant, size, className }))} ref={ref} {...props} />
    );
  }
);

Button.displayName = "Button";
