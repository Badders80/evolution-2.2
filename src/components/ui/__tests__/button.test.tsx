import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Button } from "../button";

describe("Button", () => {
  it("renders default variant", () => {
    const { asFragment } = render(<Button>Click me</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders outline variant", () => {
    const { asFragment } = render(<Button variant="outline">Click me</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders glow variant", () => {
    const { asFragment } = render(<Button variant="glow">Click me</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders large size", () => {
    const { asFragment } = render(<Button size="lg">Click me</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
