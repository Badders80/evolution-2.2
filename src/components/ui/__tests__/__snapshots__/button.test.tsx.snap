// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Button renders default variant 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-gold text-black hover:bg-gold/80 px-4 py-2"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Button renders glow variant 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-gold text-black shadow-lg shadow-gold/50 hover:shadow-xl hover:shadow-gold/70 px-4 py-2"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Button renders large size 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md font-medium transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-gold text-black hover:bg-gold/80 px-6 py-3 text-lg"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Button renders outline variant 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none border border-gold text-gold hover:bg-gold hover:text-black px-4 py-2"
  >
    Click me
  </button>
</DocumentFragment>
`;
